File: nodes/get-last-n-messages/get-last-n-messages.process.py
from typing import Any, Dict, List, Union

async def process(inputs: Dict[str, Any], settings: Dict[str, Any], config: Dict[str, Any], nodeConfig: Dict[str, Any]) -> Dict[str, Any]:
    try:
        messages = inputs.get('messages')
        count = inputs.get('count')

        if not isinstance(messages, list):
            raise Exception("Messages input must be an array")

        if not isinstance(count, int) or count < 0:
            raise Exception("Count must be a non-negative integer")

        # Get the last N messages
        last_messages = messages[-count:] if count > 0 else []
        actual_count = len(last_messages)

        return {
            "messages": last_messages,
            "count": actual_count
        }

    except Exception as error:
        print(f'Get Last N Messages error: {error}')
        raise Exception(f"Get Last N Messages error: {str(error)}")

File: nodes/get-last-n-messages/get-last-n-messages.process.js
export default async ({inputs, settings, config}) => {
    try {
        const messages = inputs.messages;
        const count = inputs.count;

        if (!Array.isArray(messages)) {
            throw new Error("Messages input must be an array");
        }

        if (typeof count !== 'number' || count < 0 || !Number.isInteger(count)) {
            throw new Error("Count must be a non-negative integer");
        }

        // Get the last N messages
        const lastMessages = count === 0 ? [] : messages.slice(-count);
        const actualCount = lastMessages.length;

        return {
            messages: lastMessages,
            count: actualCount
        };

    } catch (error) {
        console.log('Get Last N Messages error:', error);
        throw new Error(`Get Last N Messages error: ${error.message}`);
    }
};
